api/http-request-practice.md

```diff
@@ -1,90 +1,271 @@
-# Example Topic Outline (Replace with Title)
+# HTTP Request Practice
 
 ### Projected Time
 
-Example: 30-45 minutes
+** TODO: add after someone dry-runs it**
+Example: ????
 - Lesson: ? min
 - Guided Practice: ?-? min
 - Independent Practice: ? min
 - Check for Understanding: ?-? min
 
 ### Prerequisites
 
-Here are links to lessons that should be completed before this lesson:
-
-- Example: [Git version control](version-control/git-version-control/git-version-control.md)
+* [APIs & JSON](./apis-and-json.md)
+* [Twilio API](./twilio-api.md)
+* [REST API](./REST-API.md)
 
 ### Motivation
 
-Here's why this topic should be learned. Strengthen this section by including a company that uses this skillset, and a screenshot example if possible.
+HTTP is the most common way you will retrieve anything from the internet: usually web pages or API data (see [API & JSON lesson](https://github.com/Techtonica/curriculum/blob/master/api/apis-and-json.md)). It's the backbone of the web. Since HTTP is so simple and versatile (it's just text!), you can make requests lots of different ways.
 
 ### Objectives
 
 **Participants will be able to:**
 
-- leave this lesson knowing this
-- and how to do this on their own
-- Even more things
-- Even more things
+- Make HTTP requests using a variety of mechanisms.
 
 ### Specific Things to Learn
 
-- Focus on this specific talking point
-- and this
-	- This is a sub-thing about the thing
-- Even more things about the things
-- Even more things about the things
+- Basics of an HTTP Request
+- Six different ways to make an HTTP request
+	- Postman (Mac GUI app)
+	- curl
+	- XMLHttpRequest (aka AJAX) in the browser
+	- fetch (promises) in the browser
+	- http.request in Node.js
+	- fetch in Node.js
 
 ### Materials
 
-- [Example video (10 min)](https://example.com) - Write a very short description of this resource.
-- [Other example article(20 min read)](https://otherexample.com) - Answer the 3 questions at the end.
-(Be sure to explain how to use these materials in directions in one of the sections below (ie: lesson, guided practice).  If you are not assigning them somehow, they should be placed under Supplemental Materials at the bottom.) 
+- [HTTP Crash Course](https://www.youtube.com/watch?v=iYM2zFP3Zn0) - _first 17 minutes are all that matter but feel free to watch the whole thing as it covers Postman_
+-  b0rk's [Anatomy of an HTTP Request](https://twitter.com/b0rk/status/1145362860136177664/photo/1)
 
 ### Lesson
 
-Here's text about introducing something and how it works.
+HTTP requests are text-based messages used to request a resource from a server. In these lessons we'll focus on GET requests but there are other methods as well, such as POST, which is used on a form to submit data or in an API to save data.
+
+- [Google Slides](https://docs.google.com/presentation/d/1UI4XmRD3usA67ddO_8VLU3vAR4zfFXWZZaiUSKNLCXM/edit#slide=id.p)
```

Comment:
@alodahl I added these slides if you have a chance to review them

Reply:
Maybe I’m just sleepy, but your first few slides are killin me 😹

Reply:
That looks great!!